weather-api-widget/weather.lua

```diff
@@ -15,34 +15,44 @@ local gears = require("gears")
 local beautiful = require("beautiful")
 
 local HOME_DIR = os.getenv("HOME")
-local WIDGET_DIR = HOME_DIR .. '/.config/awesome/awesome-wm-widgets/weather-widget'
+local WIDGET_DIR = HOME_DIR .. '/.config/awesome/awesome-wm-widgets/weather-api-widget'
 local GET_FORECAST_CMD = [[bash -c "curl -s --show-error -X GET '%s'"]]
 
 local SYS_LANG = os.getenv("LANG"):sub(1, 2)
 if SYS_LANG == "C" or SYS_LANG == "C." then
     -- C-locale is a common fallback for simple English
     SYS_LANG = "en"
 end
--- default language is ENglish
-local LANG = gears.filesystem.file_readable(WIDGET_DIR .. "/" .. "locale/" ..
-                                      SYS_LANG .. ".lua") and SYS_LANG or "en"
-local LCLE = require("awesome-wm-widgets.weather-widget.locale." .. LANG)
--- WeatherAPI supports only these according to https://www.weatherapi.com/docs/
--- ar, bn, bg, zh, zh_tw, cs, da, nl, fi, fr, de, el, hi, hu, it, ja, jv, ko,
--- zh_cmn, mr, pl, pt, pa, ro, ru, sr, si, sk, es, sv, ta, te, tr, uk, ur, vi,
--- zh_wuu, zh_hsn, zh_yue, zu
-
 
-local function show_warning(message)
+local function show_warning(message, locale)
     naughty.notify {
         preset = naughty.config.presets.critical,
-        title = LCLE.warning_title,
+        title = locale.warning_title,
```

Comment:
Hm, how would this behave on an unsupported locale? (Say, `es` for a Spanish)

Reply:
Ah, wait. I confused `lang` (the language code) with `locale` (the translated table) here. My bad.